src/black/__init__.py

```diff
@@ -5174,35 +5176,78 @@ def normalize_numeric_literal(leaf: Leaf) -> None:
     in Python 2 long literals).
     """
     text = leaf.value.lower()
-    if text.startswith(("0o", "0b")):
+    if is_oct_or_binary(text):
         # Leave octal and binary literals alone.
         pass
-    elif text.startswith("0x"):
```

Comment:
I don't think we need this refactoring; to me it just creates an excessive number of small functions.

Reply:
That is fair - I relied on the Uncle Bob principle of "If you need a comment, then you should probably just have a more descriptive function name", but i might be misquoting or misunderstanding that. It is definitely a subjective matter. I liked the, in my opinion, higher level of readability, but the, already long, file gets longer which is maybe not cool. We can also drop the whole refactoring commit (or only take the formatting parts).

It's subjective :) I'll go with what you guys prefer.

Reply:
It doesn't hurt to add a comment to some of these if statements. E.g.:

`elif "e" in text:  # scientific notation`

Reply:
The function, in my opinion, does exactly the same without duplication and without revealing implementation details. Is a general consesus that the functions decrease quality? As mentioned, i explicitly added it in a seperate commit to make it easy to drop :)